src/editors/containers/VideoEditor/components/VideoSettingsModal/components/VideoSourceWidget/index.jsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { connect, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import {
+  Form,
+  IconButton,
+  Icon,
+  OverlayTrigger,
+  Tooltip,
+  Button,
+} from '@edx/paragon';
+import { Delete, Info, Add } from '@edx/paragon/icons';
+import {
+  FormattedMessage,
+  injectIntl,
+  intlShape,
+} from '@edx/frontend-platform/i18n';
+
+import hooks from '../hooks';
+import * as module from './hooks';
+import messages from './messages';
+import { actions } from '../../../../../../data/redux';
+
+import CollapsibleFormWidget from '../CollapsibleFormWidget';
+
+/**
+ * Collapsible Form widget controlling video source as well as fallback sources
+ */
+export const VideoSourceWidget = ({
+  // error,
+  // injected
+  intl,
+  // redux
+  updateField,
+}) => {
+  const dispatch = useDispatch();
+  const {
+    videoSource: source,
+    fallbackVideos,
+    allowVideoDownloads: allowDownload,
+  } = hooks.widgetValues({
+    dispatch,
+    fields: {
+      [hooks.selectorKeys.videoSource]: hooks.genericWidget,
+      [hooks.selectorKeys.fallbackVideos]: hooks.arrayWidget,
+      [hooks.selectorKeys.allowVideoDownloads]: hooks.genericWidget,
+    },
+  });
+  const deleteFallbackVideo = module.deleteFallbackVideo({ fallbackVideos: fallbackVideos.formValue, dispatch });
+
+  return (
+    <CollapsibleFormWidget
+      title={intl.formatMessage(messages.titleLabel)}
+    >
+      <Form.Group>
+        <div className="border-primary-100 border-bottom pb-4">
+          <Form.Control
+            floatingLabel={intl.formatMessage(messages.videoIdOrUrlLabel)}
+            onChange={source.onChange}
+            onBlur={source.onBlur}
+            value={source.local}
+          />
+        </div>
+        <Form.Label className="mt-3">
+          <FormattedMessage {...messages.fallbackVideoTitle} />
+        </Form.Label>
+        <Form.Text>
+          <FormattedMessage {...messages.fallbackVideoMessage} />
+        </Form.Text>
+        {fallbackVideos.formValue.map((videoUrl, index) => (
+          <Form.Row className="mt-4.5">
+            <Form.Control
+              floatingLabel={intl.formatMessage(messages.fallbackVideoLabel)}
+              onChange={fallbackVideos.onChange(index)}
+              value={fallbackVideos.local[index]}
+              onBlur={fallbackVideos.onBlur(index)}
+            />
+            <OverlayTrigger
```

Comment:
What does an overlay trigger do here? I don't know much about this component and can't find it in the paragon docs

Reply:
`OverlayTrigger` controls the `Tooltip` component. It is defined in the `Overlay` paragon folder, which is based on [react-bootstrap](uhttps://react-bootstrap.github.io/components/overlays/#overlaytriggerrl)